src/ldp/http/AcceptPreferenceParser.ts

```diff
@@ -40,6 +41,13 @@ export class AcceptPreferenceParser extends PreferenceParser {
       result.datetime = [{ value: input.headers['accept-datetime'] as string, weight: 1 }];
     }
 
+    if (!result.type) {
+      result.type = [];
+    }
```

Comment:
Any reason why we are not doing this above? We seem to have the exact opposite condition there: https://github.com/solid/community-server/pull/316/files#diff-87f33fa25d98a6c736238125a4aa2ac23945eae2bb52e4a99a49751a318c5504R34-R36

Reply:
The reason is I just added it there to make sure there is always an array to add the default preference to. But it is true that so far we have not clearly defined what the difference is between having an empty array and not having the preference field present at all.